metrics/cgroups/metrics.go

```diff
@@ -93,6 +97,41 @@ func (c *collector) collect(id, namespace string, cg cgroups.Cgroup, ch chan<- p
 	}
 }
 
+func (c *collector) fakecollect(ch chan<- prometheus.Metric, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for key, stored := range c.finalMetrics {
```

Comment:
Delete this function and move this logic into `Collect`.  It should flush any in-memory metrics to the channel while the go routines are running to collect new metrics.